File: backend/schemas/reports.py
"""
schemas/reports.py
Fleet Management System — Phase 8
"""

from __future__ import annotations
from typing import Optional
from schemas.common import CamelBase


class TripsSummaryReport(CamelBase):
    total_trips:       int
    pending:           int
    en_route:          int    # → enRoute
    completed:         int
    cancelled:         int
    total_distance_km: float
    avg_duration_hours: float


class MaintenanceSummaryReport(CamelBase):
    total_work_orders: int
    pending:           int
    in_progress:       int   # → inProgress
    completed:         int
    overdue:           int
    total_cost:        float
    avg_cost:          float
    currency:          str


class DriverPerformanceRow(CamelBase):
    driver_id:          str
    driver_name:        str
    total_trips:        int
    completed_trips:    int
    cancelled_trips:    int
    total_distance_km:  float
    on_time_rate:       float   # 0.0–1.0, → onTimeRate


class DriverPerformanceReport(CamelBase):
    drivers: list[DriverPerformanceRow]
    date_from: Optional[str] = None
    date_to:   Optional[str] = None

File: frontend/web/src/types/reports.ts
/**
 * types/reports.ts
 * Fleet Management System — Phase 8
 */

export interface TripsSummaryReport {
  totalTrips:       number;
  pending:          number;
  enRoute:          number;
  completed:        number;
  cancelled:        number;
  totalDistanceKm:  number;
  avgDurationHours: number;
}

export interface MaintenanceSummaryReport {
  totalWorkOrders: number;
  pending:         number;
  inProgress:      number;
  completed:       number;
  overdue:         number;
  totalCost:       number;
  avgCost:         number;
  currency:        string;
}

export interface DriverPerformanceRow {
  driverId:        string;
  driverName:      string;
  totalTrips:      number;
  completedTrips:  number;
  cancelledTrips:  number;
  totalDistanceKm: number;
  onTimeRate:      number;
}

export interface DriverPerformanceReport {
  drivers:  DriverPerformanceRow[];
  dateFrom?: string;
  dateTo?:   string;
}

export interface ReportDateParams {
  dateFrom?: string;
  dateTo?:   string;
}